### ISSUE TITLE
Feature Request: Add weakness array to temtem data

### ISSUE THREAD

**Issue by BaileyMillerSSI** [maael/temtem-api#68]
On the wiki there is a type matchup chart, would it be possible to include this field in the temtem data. It is currently very weird to lookup this data using the existing api endpoints. For example given any random temtem I want to have a field to pull the type matchup from highest modifier to lowest. 

![image](https://user-images.githubusercontent.com/14271740/106977787-fa434f00-6728-11eb-9c7e-70c792df73aa.png)

**Comment by BaileyMillerSSI** [comment]
For reference I am working on a discord bot and I want to be able to display the best type to attack a specific Tem with. I am currently figuring this out by pulling the Tem's type and the entire weakness data and then looping over that and creating a list of type and modifier.

**Comment by maael** [comment]
Hey there again 👋 , thanks for the suggestion.

This has been added now, but it won't be included by default, and is behind a flag - as I'm having bandwidth issues with this project already, serving 100,000s of requests each month, and don't want everything doing requests to have this added object per temtem too.

To access this info, you can either you the temtem list or specific endpoints, and add `weaknessess=1` as a query param, for example:

For the full list: https://temtem-api.mael.tech/api/temtems?weaknesses=1
For a specific temtem: https://temtem-api.mael.tech/api/temtems/133?weaknesses=1

And the resulting object is something like this on each temtem:

```json
{
  ...rest of temtem data,
  "weaknesses": {
    "Neutral":1,
    "Fire":2,
    "Water":1,
    "Nature":1,
    "Electric":1,
    "Earth":1,
    "Mental":0.5,
    "Wind":0.5,
    "Digital":1,
    "Melee":2,
    "Crystal":1,
    "Toxic":0.5
  }
}
```

Where the key of the object is the attacking type, and the value is the modifier when that type is used to attack the specified temtem (taking into account all types on that temtem). I hope that all makes sense.

I think that solves your use case, so I'll close this issue - please reopen it or open a new issue if you think there's more to it though. 👍

**Comment by BaileyMillerSSI** [comment]
Hey @maael I am not sure how you're hosting this but I just setup my bot to run using Digital Ocean apps. That may be something to consider depending on the cost/performance/bandwidth limits. 
https://www.digitalocean.com/products/app-platform/

**Comment by maael** [comment]
Thanks for the suggestion, I do run a number of things on DigitalOcean and Vercel. This is on Vercel at the moment, behind Cloudflare with caching rules set up as the data is only refreshed every 4 hours. I've also made sure to minimize and compress all the images included in the project too. I managed to get it down to 10% of the original bandwidth from before the optimisations.

My general policy is that for something that makes no money like this, I prefer to not have any costs for hosting it, so the £5/mo isn't something I really entertain right now.
